### PR TITLE
dodo: init at 2022-08-13

### PR DESCRIPTION
**Pull request by matthiasbeyer**
Maybe someone can help here. The app does not start because incompatible qt versions. I am not a python nor a qt packager, so I don't know what is happening or how to investigate...

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/mailreaders/dodo/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{ pkgs, lib, python3 }:
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "dodo";
+  version = "2022-08-13";
```
Comment (azahi) [inline pkgs/applications/networking/mailreaders/dodo/default.nix:5]: This should be `unstable-YYYY-MM-DD` where YYYY-MM-DD is the date of the commit.

File: pkgs/applications/networking/mailreaders/dodo/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    bleach
+    pyqt5
+    pyqt5_sip
+    pyqtwebengine
+  ];
+
+  meta = with lib; {
+    description = "A graphical, hackable email client based on notmuch ";
+    homepage = "https://github.com/akissinger/dodo";
+    license = licenses.gpl3;
```
Comment (azahi) [inline pkgs/applications/networking/mailreaders/dodo/default.nix:27]: Please consult [this](https://discourse.nixos.org/t/lib-licenses-gpl3-co-are-now-deprecated/8206) and change the license attribute accordingly.

File: pkgs/applications/networking/mailreaders/dodo/default.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+  propagatedBuildInputs = with python3.pkgs; [
+    bleach
+    pyqt5
+    pyqt5_sip
+    pyqtwebengine
+  ];
+
+  meta = with lib; {
+    description = "A graphical, hackable email client based on notmuch ";
```
Comment (azahi) [inline pkgs/applications/networking/mailreaders/dodo/default.nix:25]:
```suggestion
    description = "A graphical, hackable email client based on notmuch";
```

**Review by azahi — CHANGES_REQUESTED** [review]
I think you need to wrap the executable with [prerequisites](https://github.com/akissinger/dodo#prerequisites), otherwise it won't work for if, for instance, `notmuch` is not installed.

File: pkgs/applications/networking/mailreaders/dodo/default.nix [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "dodo";
+  version = "2022-08-13";
+  format = "pyproject";
+
+  src = pkgs.fetchFromGitHub {
+    repo = pname;
+    owner = "akissinger";
+    rev = "b377e05ab507c7bec427e3bac0c35b68bd67d9fc";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/mailreaders/dodo/default.nix:11]:
```suggestion
  src = fetchFromGitHub {
    owner = "akissinger";
    repo = "dodo";
    rev = "b377e05ab507c7bec427e3bac0c35b68bd67d9fc";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by matthiasbeyer** [comment]
> I think you need to wrap the executable with [prerequisites](https://github.com/akissinger/dodo#prerequisites), otherwise it won't work for if, for instance, notmuch is not installed.

I agree, at least for w3m. For notmuch not so much, because a user must configure it anyways, so they know that it must be installed. Not sure whether we should just "imply" it in this package.

**Comment by azahi** [comment]
> > I think you need to wrap the executable with [prerequisites](https://github.com/akissinger/dodo#prerequisites), otherwise it won't work for if, for instance, notmuch is not installed.
> 
> I agree, at least for w3m. For notmuch not so much, because a user must configure it anyways, so they know that it must be installed. Not sure whether we should just "imply" it in this package.

I still think that notmuch should be included because it is essential for the program to work and it would be expected that we ship a program with all required components included, so that it could be run via `nix run nixpkgs#dodo` without hiccups. A user may override the package if he chooses to use a custom build of notmuch. Although, we can ask for other opinions on this matter if you'd like.

@SuperSandro2000 ^

**Comment by matthiasbeyer** [comment]
> I still think that notmuch should be included because it is essential for the program to work and it would be expected that we ship a program with all required components included, so that it could be run via nix run nixpkgs#dodo without hiccups. A user may override the package if he chooses to use a custom build of notmuch

Yes, I think that is reasonable! :+1:

**Comment by matthiasbeyer** [comment]
@azahi please have a look at the latest patch, if that's the correct way. If yes, I will squash these commits into one.

**Review by azahi — APPROVED** [review]

**Comment by matthiasbeyer** [comment]
Closing as I don't care enough and this has been open for way too long.

**Pull request closed without merging**